lib/ansible/modules/system/alternatives.py

```diff
@@ -132,40 +130,93 @@ def main():
             # This is only compatible on Debian-based systems, as the other
             # alternatives don't have --query available
             rc, query_output, _ = module.run_command(
-                ['env', 'LC_ALL=C', UPDATE_ALTERNATIVES, '--query', name]
+                ['env', 'LC_ALL=C', cmd, '--query', name]
             )
             if rc == 0:
                 for line in query_output.splitlines():
                     if line.startswith('Link:'):
                         link = line.split()[1]
                         break
 
+    return current_path, all_alternatives, link
+
+def set_alternative(module, cmd, name, path, link, priority,
+        current_path, all_alternatives):
+    """Set the current named alternative to the specified path.
+
+    If not present in the currently configured alternatives, the requested
+    alternative will be added with the specified priority.  The `link`
+    parameter must be provided in this instance."""
     if current_path != path:
         if module.check_mode:
             module.exit_json(changed=True, current_path=current_path)
         try:
             # install the requested path if necessary
             if path not in all_alternatives:
                 if not link:
-                    module.fail_json(msg="Needed to install the alternative, but unable to do so as we are missing the link")
+                    module.fail_json(msg="Needed to install the alternative, "
+                        "but unable to do so as we are missing the link")
 
                 module.run_command(
-                    [UPDATE_ALTERNATIVES, '--install', link, name, path, str(priority)],
+                    [cmd, '--install', link, name, path, str(priority)],
                     check_rc=True
                 )
 
             # select the requested path
             module.run_command(
-                [UPDATE_ALTERNATIVES, '--set', name, path],
+                [cmd, '--set', name, path],
                 check_rc=True
             )
 
             module.exit_json(changed=True)
         except subprocess.CalledProcessError:
-            e = get_exception()
+            cpe = get_exception()
             module.fail_json(msg=str(dir(cpe)))
     else:
         module.exit_json(changed=False)
 
+def remove_alternative(module, cmd, name, path, all_alternatives):
+    """Remove the requested path if necessary"""
+    if path in all_alternatives:
+        try:
+            module.run_command([cmd, '--remove', name, path], check_rc=True)
+            module.exit_json(changed=True)
+        except subprocess.CalledProcessError:
```

Comment:
@alikins should I remove the try/except blocks around the calls to module.run_command()?